code/modules/antagonists/traitor/equipment/Malf_Modules.dm

```diff
@@ -859,5 +859,36 @@ GLOBAL_LIST_INIT(blacklisted_malf_machines, typecacheof(list(
 	if(AI.eyeobj)
 		AI.eyeobj.relay_speech = TRUE
 
+
+//Fake Alert: Overloads a random number of lights across the station. Three uses.
+/datum/AI_Module/small/fake_alert
+	module_name = "Fake Alert"
+	mod_pick_name = "fake_alert"
+	description = "Assess the most probable threats to the station, and send a distracting fake alert by hijacking the station's alert and threat identification systems."
+	cost = 20
+	power_type = /datum/action/innate/ai/fake_alert
+	unlock_text = "<span class='notice'>You gain control of the station's alert system.</span>"
+	unlock_sound = "sparks"
+
+/datum/action/innate/ai/fake_alert
+	name = "Fake Alert"
+	desc = "Scare the crew with a fake alert."
+	button_icon_state = "fake_alert"
+	uses = 1
+
+/datum/action/innate/ai/fake_alert/Activate()
+	var/list/events_list = list()
+	for(var/datum/round_event_control/E in SSevents.control)
+		if(!E.malffake)
```

Comment:
it doesn't follow naming scheme and seems like bad approach to this. Check for weight (0 should mean event is disabled).

Reply:
Like, running through all events and listing the ones with weight>0?

Reply:
yes

Reply:
also this list should be static and generated once

Reply:
static lists have downsides too, for example if this is only called once during the round, that bit of memory will be occupied for the entire game. If it's something that'll rarely be called it's worth weighing if the gains are greater than the losses

Reply:
Not sure this is the best approach, as it will list events that don't have alerts (heart attack, mass halucination).